bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -646,7 +646,7 @@ protected void unsetManagedThingProvider(ManagedThingProvider thingProvider) {
         this.managedThingProvider = null;
     }
 
-    @Reference(policy = ReferencePolicy.DYNAMIC)
+    @Reference(policy = ReferencePolicy.DYNAMIC, target="(type=temporaryStorage)")
```

Comment:
Nice. Now see if we can’t do the same thing without modifying the ESH sources, just using configuration.

A Component can be configured via its `configurationPid`. If unset, it is set to the same value as the component `name`. If _that_ is unset, it uses the fully qualified name of the annotated class.

What I’ve read leads me to believe that we can use that to set a target for this reference using a configuration parameter like this:
```
org.eclipse.smarthome.config.discovery.internal.PersistentInbox:storageService.target
```

Does that work?